README.md

```diff
@@ -5,33 +5,32 @@ This package implements are generic user interface to popular event generators u
 Simulate interactions with one of the supported event generators 
 
 ```python
-from impy.definitions import *
-from impy.constants import *
-from impy.kinematics import EventKinematics
-from impy import impy_config
+import numpy as np
+import impy
+from impy.constants import TeV
 
 # Define the parameters of the collisions
-event_kinematics = EventKinematics(
-    ecm=13 * TeV, p1pdg=2212, p2pdg=2212)
+event_kinematics = impy.kinematics.EventKinematics(ecm=13 * TeV, p1pdg=2212, p2pdg=2212)
+# Create an instance of an event generator
+generator = impy.models.Sibyll23d(event_kinematics)
 
-# Create an instance of an event generator by passing
-# the model name as a string
-generator = make_generator_instance(
-    interaction_model_by_tag['SIBYLL2.3D'])
+nevents = 0
+average_pt = 0
 
-# Initialize it
-generator.init_generator(event_kinematics)
-
-# Number of events to generate
-nevents = 100
-
-for event in generator.event_generator(event_kinematics, nevents):
-    event.filter_final_state_charged()
+# Generate 10000 events
+for event in generator(10000):
+    # Filter events
+    event.filter_final().filter_charged()
     # do something with event.p_ids, event.eta, event.en, event.pt, etc.
     # these variables are numpy arrays, that can be histogrammed or counted like
-    average_pt += 1/float(nevents)*np.mean(event.pt[np.abs(event.p_ids) == 211])
-
-print('Average pT for charged pions {0:4.3f}'.format(average_pt))
+    pt = event.pt[np.abs(event.p_ids) == 211]
+    # The list could be empty
+    if len(pt) > 0:
+        nevents += 1
+        average_pt += np.mean(pt)
+
+average_pt = 1 / float(nevents) * average_pt
```

Comment:
While this line is faster than `average_pt / nevents`, I would keep the code as simple as possible in a pedagogical example, because the difference in speed is imperceptible. I suggest to change the line to `average_pt = average_pt / nevents`. This will work as intended, no `float` needed.

This does not work: `average_pt /= nevents`, because it is an in-place operation. The numpy devs decided that this should not change the dtype of the array.

Reply:
Changed.